actions/ulog_ingestion/src/ulog_ingestion/__main__.py

```diff
@@ -0,0 +1,295 @@
+    )
+
+    relative_file_name = output_path_per_topic_mcap.split(output_dir_temp)[1][1:]
+
+    # Upload MCAP File
+    dataset.upload_file(
+        pathlib.Path(output_path_per_topic_mcap), relative_file_name
+    )
+
+    print(
+        f"Setting default representation for topic: {topic_name_roboto}, file_id: {dataset.get_file_info(relative_file_name).file_id}"
+    )
+
+    # Set Default Topic Representation
+    topic.set_default_representation(
+        request=topics.SetDefaultRepresentationRequest(
+            association=Association(
+                association_type=AssociationType.File,
+                association_id=dataset.get_file_info(relative_file_name).file_id,
+            ),
+            org_id=org_id,
+            storage_format=topics.RepresentationStorageFormat.MCAP,
+            version=1,
+        )
+    )
+    return
+
+
+def ingest_ulog(ulog_file_path: str, topics: List[str] = None):
+    """
+
+    This function creates topic entries, message path records, and MCAP files from a ULog file.
+
+    Args:
+    - ulog_file_path: Path to the .ulg file.
+    - messages: List of messages to process, separated by commas.
+
+    Returns:
+    - None
+    """
+    msg_filter = topics.split(",") if topics else None
+    print(ulog_file_path)
+
+    ulog = ULog(ulog_file_path, msg_filter, True)
+
+    _, input_dir, _, _, _ = setup_env()
+
+    output_dir_path_mcap, output_dir_temp = utils.setup_output_folder_structure(
+        ulog_file_path, input_dir
+    )
+    start_time = time.time()
+
+    # This is a temporary fix for Multi Information messages with the same name
+    # https://docs.px4.io/main/en/dev_log/ulog_file_format.html#m-multi-information-message
+    # Similar to Plotjuggler approach
+    # TODO: handle this in a better way
+    message_names_with_multi_id_list = list()
+    for d in sorted(ulog.data_list, key=lambda obj: obj.name):
+        print(f"topic name: {d.name}")
+        if d.multi_id > 0:
+            message_names_with_multi_id_list.append(d.name)
+
+    message_names_with_multi_id_list = list(set(message_names_with_multi_id_list))
+
+    print(f"message_names_with_multi_id_list: {message_names_with_multi_id_list}")
+
+    # Prepare Arguments for Parallel Processing
+    args_list = [
+        (
+            ulog_file_path,
+            ulog_data_object,
+            ulog.message_formats,
+            message_names_with_multi_id_list,
+            output_dir_path_mcap,
+            output_dir_temp,
+        )
+        for ulog_data_object in sorted(ulog.data_list, key=lambda obj: obj.name)
+    ]
+
+    with ProcessPoolExecutor() as executor:
+        # Submit all tasks and collect Future objects
+        futures = [executor.submit(process_data, *args) for args in args_list]
+
+        for future in as_completed(futures):
+            try:
+                result = future.result()
+                print("Task completed successfully", result)
```

Comment:
Minor: `process_data` returns `None`, so this is printing `Task completed successfully None`. 

Something I like to do when I use `concurrent.futures.as_completed` is to pass it a mapping instead of a list, so that you can lookup info about the future and print meaningful logs. The docs have a good example of that -- here's the important part:
```python
future_to_url = {executor.submit(load_url, url, 60): url for url in URLS}
for future in concurrent.futures.as_completed(future_to_url):
    url = future_to_url[future]
    try:
        data = future.result()
    except Exception as exc:
        print('%r generated an exception: %s' % (url, exc))
    else:
        print('%r page is %d bytes' % (url, len(data)))
```
From https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor-example